### PR TITLE
test: fix data race in status server

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31999

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
None
```

### CODE REVIEW

File: server/http_status.go [hunk 1]
```diff
@@ -453,24 +453,27 @@ func (s *Server) startStatusServerAndRPCServer(serverMux *http.ServeMux) {
...
+		err := statusServer.Serve(httpL)
 		logutil.BgLogger().Error("http server error", zap.Error(err))
 	}, nil)
 
 	err := m.Serve()
 	if err != nil {
 		logutil.BgLogger().Error("start status/rpc server error", zap.Error(err))
 	}
+
+	s.statusServer = statusServer
```
Comment (Defined2014) [inline server/http_status.go:475]: Does it cause grpc server stop failed?

**Review by Defined2014 — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**